generated/nidcpower/nidcpower/lcr_load_compensation_spot.py

```diff
@@ -0,0 +1,116 @@
+            ideal_capacitance, ideal_inductance and ideal_resistance can be set, and the remaining
+            parameters (excluding frequency) must be None. The parameter that is not None specifies
+            the known specification value of the DUT to be used as the basis for load compensation.
+            If all of them are None, then the default value of `impedance=complex()` will be used.
+
+        Args:
+            frequency (float): Specifies the spot frequency, in Hz.
+
+            impedance (complex): Specifies the actual impedance of your DUT to be used as the basis
+                for load compensation, or None to use another type of DUT specification value.
+
+            ideal_capacitance (float): Specifies the ideal capacitance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_inductance (complex): Specifies the ideal inductance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+
+            ideal_resistance (complex): Specifies the ideal inductance of your DUT to be used as the
+                basis for load compensation, or None to use another type of DUT specification value.
+        """
+        self.frequency = frequency
+        # Set default values
+        self.reference_value_type = enums.LCRReferenceValueType.IMPEDANCE
+        self.reference_value = complex()
+        # Input validations
+        none_count = 0
+        for uppercase_parameter_name in enums.LCRReferenceValueType.__members__:
+            parameter_name = uppercase_parameter_name.lower()
+            parameter = locals()[parameter_name]
+            if parameter is None:
+                none_count += 1
+            elif parameter_name == "impedance" and not isinstance(parameter, numbers.Complex):
+                raise TypeError("Parameter impedance must be of type complex")
+            elif parameter_name != "impedance" and not isinstance(parameter, numbers.Real):
+                raise TypeError("Parameter {} must be a real number".format(parameter_name))
+            else:
+                self.reference_value_type = getattr(
+                    enums.LCRReferenceValueType, uppercase_parameter_name
+                )
+                self.reference_value = parameter
+
+        if none_count < len(enums.LCRReferenceValueType) - 1:
+            raise ValueError(
+                "At most one of {0} parameters can be set and the remaining parameters must be None".format(
+                    tuple(
+                        uppercase_parameter_name.lower()
+                        for uppercase_parameter_name in enums.LCRReferenceValueType.__members__
+                    )
+                )
+            )
+
+    def __repr__(self):
+        return "{0}(frequency={1}, {2}={3})".format(
+            self.__class__.__name__,
```

Comment:
```suggestion
        return "{0}.{1}(frequency={2}, {3}={4})".format(
            self.__class__.__module__,
            self.__class__.__qualname__,
```

I think you need to do like above to be able to recreate class instances based on `repr` output.

Reply:
Thanks for the suggestion, I have updated it in my latest commit.